### PR TITLE
Feat#1 setup storybook

### PR DESCRIPTION
**Pull request by OleDakotaJoe**
# Summary
Set up the project with storybook, as well as implemented a simple component

## What changed?
Project was empty before, now it has a basic structure

## Why was it necessary?
This is a foundational step towards getting the project set up.

## What testing was completed? If none, why?
No testing was implemented because there is currently no testing framework implementation. 
Also, this is to allow someone else the opportunity to implement the testing framework, as well as write a test for this component.

### CODE REVIEW

File: src/stories/Button.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    },
+} as Meta<typeof Button>;
+
+const Template: Story<ButtonProps> = (args) => <Button {...args} />;
+
+export const Primary = Template.bind({});
+Primary.args = {
+    primary: true,
+    disabled: false,
+    text: 'Primary',
```
Comment (mdmathewdc) [inline src/stories/Button.stories.tsx:19]: We should stick with Double quotes or single quotes throughout the whole project..I prefer single quotes but I think we should change to make it more consistent in all files.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    "@storybook/addon-actions": "^6.4.19",
+    "@storybook/addon-essentials": "^6.4.19",
+    "@storybook/addon-interactions": "^6.4.19",
+    "@storybook/addon-links": "^6.4.19",
+    "@storybook/react": "^6.4.19",
+    "@storybook/testing-library": "^0.0.9",
+    "@types/react": "^17.0.39",
+    "@types/styled-components": "^5.1.23",
+    "babel-loader": "^8.2.3",
+    "react": "^17.0.2",
```
Comment (aetheode) [inline package.json:46]: We need to add the react-dom library

File: src/stories/Button.stories.tsx (hunk 1 above)
Comment (OleDakotaJoe) [reply src/stories/Button.stories.tsx:19]: Yea, copy pasta my friend. 
We should probably create a style guidelines sheet and outline this in the CONTRIBUTING.md

Also, I like single quotes as well. 

You've already got some experience as an engineer, I bet. Good catch.

Comment (Ayon95) [reply src/stories/Button.stories.tsx:19]: Perhaps we can add a Prettier config file for consistent formatting.

**Comment by OleDakotaJoe** [comment]
Yea definitely, I would say go ahead and add an issue to add prettier, and then can wrote that as an expectation in contributing, that you bring the formatter before checking in code

Thanks,
Steven Sheaves
(931) 436-5385
________________________________

File: src/components/Button.component.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import {ButtonProps} from './Button.types'
+import {ButtonColor, ButtonSizes} from "./Button.constants";
+
+const StyledButton = styled.button<ButtonProps>`
+  
+    position: relative;
+	padding: .7em 1.4em;
+	width: fit-content;
+	font-size: ${props => props.size == ButtonSizes.SMALL ? '10px' : (props.size == ButtonSizes.MEDIUM ? '15px' : '20px' )  };
+	color: ${props => props.primary ? props.secondaryColor : props.primaryColor};
```
Comment (mdmathewdc) [inline src/components/Button.component.tsx:13]: It should be color: ${props => props.primary ? props.primaryColor : props.secondaryColor };

File: src/components/Button.component.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	background-color: ${props => props.primary ? props.primaryColor : 'transparent'};
+	border: solid 2px  ${props => props.primary ?  props.secondaryColor :  props.primaryColor};
```
Comment (mdmathewdc) [inline src/components/Button.component.tsx:15]: Here too...I think you meant the opposite.

**Comment by Nelson-Chinedu** [comment]
Instead of using relative import statement like `.../` it would be nice to use absolute import like `src/` by setting `baseUrl: '.'` in tsconfig

**Comment by OleDakotaJoe** [comment]
Hey, you left your /.idea folder, we need add that to .gitignore

**Pull request merged**